site/public/js/gradeable-notebook.js

```diff
@@ -94,43 +129,47 @@ function saveNotebookToLocal() {
-                $(element).prop('checked', values.includes(element.value)).change();
+        //restore multiple choice
+        for (const id in inputs.multiple_choice) {
+            const filename = inputs.multiple_choice[id][0];
+            const answers = inputs.multiple_choice[id][1];
+            $('.multiple_choice').each(function () {
+                $(this).children('fieldset').each(function () {
+                    //match file names
+                    if ($(this).attr('name') === filename) {
+                        $(this).children('label').each(function () {
+                            $(this).children('input').each(function () {
+                                //check proper inputs
+                                for (let i = 0; i < answers.length; i++) {
+                                    if ($(this)[0].defaultValue === answers[i]) {
+                                        $(this).prop('checked', true);
+                                    }
+                                }
+                            });
+                        });
+                    }
+                });
             });
         }
-        // Next, we restore short-answer boxes
-        for (const id in state.short_answer) {
-            const answer = state.short_answer[id][0];
-            // Restore the answer and trigger change events (for setting button states)
-            // (see https://stackoverflow.com/questions/4672505/why-does-the-jquery-change-event-not-trigger-when-i-set-the-value-of-a-select-us)
-            $(`#${id}`).val(answer).trigger('input');
-        }
-        // Finally, we restore codeboxes
-        for (const id in state.codebox) {
-            const answer = state.codebox[id][0];
-            const codebox = $(`#${id} .CodeMirror`).get(0);
-            // If this box no longer exists, then don't attempt to update the
-            // answer. The autosave data is probably for an older version of
-            // the gradeable at this point, see issue #5351.
-            if (!codebox) {
-                continue;
-            }
-            const cm = codebox.CodeMirror;
-            // This automatically triggers the event handler for the clear and
-            // recent buttons.
-            cm.setValue(answer);
+        //restore short answers
+        for (const id in inputs.short_answer) {
+            const filename = inputs.short_answer[id][0];
+            const answers = inputs.short_answer[id][1];
+            $('.short_answer').each(function(){
+                $(this).children('div').each(function(){
+                    //match file names
+                    if ($(this).attr('class') === filename) {
+                        //set input
+                        const editor = ($(this)[0]).querySelector('.CodeMirror').CodeMirror;
+                        editor.setValue(answers);
+                    }
+                });
+            });
```

Comment:
```suggestion
            $(`.short_answer > div[data-filename="${filename}"]`).each(function(){
              //set input
              const editor = ($(this)[0]).querySelector('.CodeMirror').CodeMirror;
              editor.setValue(answers);
            });
```
This can also be condensed. This selector selects all `div` elements with the `data-filename` attribute equal to the filename that are direct children of an element with the `short-answer` class (direct children because we used `>` in between the element queries.